### DISCUSSION TITLE
Classwise mAP during validation

### DISCUSSION THREAD

**Discussion by guarin** [open-mmlab/mmyolo#761] [General]
Hi, thanks a lot for your amazing work on mmyolo!

Is it possible to show classwise mAP during validation? I currently only get the following output in the log file:
```
2023/05/03 14:58:52 - mmengine - INFO - Evaluating bbox...
2023/05/03 14:59:03 - mmengine - INFO - bbox_mAP_copypaste: 0.002 0.010 0.000 0.000 0.001 0.003
2023/05/03 14:59:04 - mmengine - INFO - Epoch(val) [1][572/572]    coco/bbox_mAP: 0.0020  coco/bbox_mAP_50: 0.0100  coco/bbox_mAP_75: 0.0000  coco/bbox_mAP_s: 0.0000  coco/bbox_mAP_m: 0.0010  coco/bbox_mAP_l: 0.0030  data_time: 0.0112  time: 0.0358
```

And in scalars.json:
```
{"coco/bbox_mAP": 0.002, "coco/bbox_mAP_50": 0.01, "coco/bbox_mAP_75": 0.0, "coco/bbox_mAP_s": 0.0, "coco/bbox_mAP_m": 0.001, "coco/bbox_mAP_l": 0.003, "data_time": 0.011173534393310546, "time": 0.035815372467041015, "step": 1}
```

I use `val_evaluator` config from `yolov7_l_syncbn_fast_8x16b-300e_coco.py`.

**Comment by hhaAndroid** [comment]
可以通过在配置中将 `test_evaluator.classwise` 设置为 True，或者在运行 `test.py` 时增加 `--cfg-options test_evaluator.classwise=True` 来打印每个类的 mAP。

**Reply by guarin** [reply]
Thanks a lot!

Just for completeness, it also works for the `val_evaluator` and can be used in validation during training.

If you save the best checkpoints, make sure to set the target metric in the checkpoint hook: `default_hooks.checkpoint.save_best="coco/bbox_mAP"`. Otherwise it will take the `mAP` of the first class in the dataset as classwise metrics are reported before overall metrics.
